### PR TITLE
fix media-gallery doesn't respect call banner

### PR DESCRIPTION
**Pull request by michaelkirk-signal**
PTAL @charlesmchen 

Came across this while testing the landscape stuff, but actually unrelated.

### CODE REVIEW

**Comment by michaelkirk-signal** [comment]
This needs some more work. Closing for now.

File: Signal/src/ViewControllers/MediaGalleryViewController.swift [hunk 1]
```diff
@@ -168,43 +168,10 @@ protocol MediaGalleryDataSourceDelegate: class {
...
-        super.init(nibName: nil, bundle: nil)
-    }
-
-    required init?(coder aDecoder: NSCoder) {
-        notImplemented()
-    }
+class MediaGalleryNavigationController: OWSNavigationController {
 
+    var presentationView: UIImageView!
+    var retainUntilDismissed: MediaGallery?
```
Comment (charlesmchen-signal) [inline Signal/src/ViewControllers/MediaGalleryViewController.swift]: I don't see this being cleared anywhere.  Will this by a retain cycle?  Are these entities being released?

Comment (michaelkirk-signal) [reply Signal/src/ViewControllers/MediaGalleryViewController.swift]: Yes, they're being released.

Ownership is basically inverted.

The MediaGallery has a *weak* reference to the MediaGalleryNavigationController it creates, which in turn has a *strong* reference back to the MediaGallery that created it.

So in effect, the lifetime of the media gallery is only as long as something *else* holds a reference to the MediaGalleryNavigationController. 

An alternative would be that everywhere we show the MediaGalleryNavigationController we also hold a reference to the MediaGallery, and then whenever we dismiss the MediaGalleryNavigationController, we also manually clear the reference to the MediaGallery.

While more explicit, that approach seemed more error prone.

I'm not crazy about the current approach, but I'm open to suggestion.

Comment (michaelkirk-signal) [reply Signal/src/ViewControllers/MediaGalleryViewController.swift]: If you prefer, I could also just revert c5bb924, and merge 97e2c33 which also works.

**Pull request merged**